core/src/main/kotlin/supplier/EntitySupplier.kt

```diff
@@ -473,6 +473,31 @@ public interface EntitySupplier {
     public suspend fun getGuildScheduledEvent(guildId: Snowflake, eventId: Snowflake): GuildScheduledEvent =
         getGuildScheduledEventOrNull(guildId, eventId) ?: EntityNotFoundException.guildScheduledEventNotFound(eventId)
 
+    public  fun getGuildScheduledEventUsersBefore(
+        guildId: Snowflake,
+        eventId: Snowflake,
+        limit: Int = Int.MAX_VALUE,
+        withMember: Boolean? = null,
+        before: Snowflake = Snowflake.max
+    ): Flow<User>
+
+
+    public fun getGuildScheduledEventUsersAfter(
+        guildId: Snowflake,
+        eventId: Snowflake,
+        limit: Int = Int.MAX_VALUE,
+        withMember: Boolean? = null,
+        after: Snowflake = Snowflake.min
+    ): Flow<User>
+
+    public fun getGuildScheduledEventUsers(
+        guildId: Snowflake,
+        eventId: Snowflake,
+        limit: Int = Int.MAX_VALUE,
+        withMember: Boolean? = null,
+    ): Flow<User> = getGuildScheduledEventUsersAfter(guildId, eventId, limit, withMember)
```

Comment:
Is there a `GuildScheduledEventUser` type? That's what the endpoint should return, see https://discord.com/developers/docs/resources/guild-scheduled-event#get-guild-scheduled-event-users

Reply:
I didn't add it yet

Reply:
It's kinda hard to decide wither this should be it's own thing or not; closed I got was the UserOptionallyMember entity 
as the given event Id is already at hand, for lower level representations they already have access to the full payload

Reply:
I would also make `before`/`after` the first parameter in their functions to be consistent with `getMessages...`, same in `Behavior`.

Reply:
Im still thinking of how to model this; the endpoint expresses a "garenteed user" but an "optional member"
However, we cache the member and provide the user

Reply:
Another idea: a second set of methods would also be an option, so you can't provide a `withMember` option but instead call another method.
But we might also just go all in for a `GuildScheduledEventUser` entity, maybe Discord will use it more in the future / add new fields to it.

Reply:
So one function returns Users; the other returns Members
Perfecto

Reply:
Done

Reply:
Also a `getGuildScheduledEventMembers()` or only for users?